history.py

```diff
@@ -0,0 +1,30 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+
+        # holds IDs
+        request.session.get('last_five', [])
```

Comment:
You might try saving the request.session to a variable so you don't have to access it again on line 15 and 16. This makes your code slower.

Reply:
Good idea! Thanks!